File: single-number.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        result = 0

        for i in nums:
            result ^= i
        return result

if __name__ == '__main__':
    s = Solution()
    nums = [1, 1, 3, 2 ,2]

    print(s.singleNumber(nums))

File: single-number/single-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
  var i;
  var result = 0;
  for (i = 0; i < nums.length; i++) {
    result ^= nums[i];
  }
  return result;
};

var main = function() {
  var nums = [1, 1, 3, 2, 2];

  console.log(singleNumber(nums));
}

if (require.main === module) {
  main();
}
